src/Core/Domain/Product/SpecificPrice/Command/EditProductSpecificPriceCommand.php

```diff
@@ -0,0 +1,406 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\SpecificPrice\Command;
+
+use DateTime;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\Domain\SpecificPrice\ValueObject\SpecificPriceId;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Reduction;
+
+class EditProductSpecificPriceCommand
+{
+    /**
+     * //@todo: replace with Product/SpecificPriceId
+     *
+     * @var SpecificPriceId
+     */
+    private $specificPriceId;
+
+    /**
+     * @var Reduction|null
+     */
+    private $reduction;
+
+    /**
+     * @var bool|null
+     */
+    private $includesTax;
+
+    /**
+     * @var DecimalNumber|null
+     */
+    private $price;
+
+    /**
+     * @var int|null
+     */
+    private $fromQuantity;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce ShopGroupIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $shopGroupId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce ShopIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $shopId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CombinationIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $combinationId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CurrencyIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $currencyId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CountryIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $countryId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce GroupIdInterface (refer to example of ManufacturerIdInterface). GroupId VO doesnt exist yet
+     */
+    private $groupId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CustomerIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $customerId;
+
+    /**
+     * @todo: its impossible to reset $from to NULL (database doesnt support null and the 0000-00-00 is actually invalid)
+     *      for that we could have some "DateRange" Value object which would contain ranges OR some "no-range" value.
+     *      Also this would require BC break - making datetime column nullable in specific_price table.
```

Comment:
question :up:

Reply:
:up:

Reply:
nvm answered myself. There is already introduced object NullDateTime, handled by that.